src/scripts/main.js

```diff
@@ -18,10 +18,34 @@ const food = {
   },
 };
 
+const treeDiv = document.createElement('div');
+
+treeDiv.id = 'tree';
+document.body.appendChild(treeDiv);
+
 const tree = document.querySelector('#tree');
 
 function createTree(element, data) {
-  // WRITE YOUR CODE HERE
+  if (element === null || data === undefined) {
+    throw new Error('arguments must not be null or undefined');
+  }
+
+  const ul = document.createElement('ul');
+
+  for (const key in data) {
+    const li = document.createElement('li');
+
+    li.textContent = key;
+
+    if (Object.entries(data[key]).length > 0) {
+      createTree(li, data[key]);
+    }
```

Comment:
Here, you are recursively calling the `createTree` function if the current key has sub-keys. However, there is no base case for the recursion. This could potentially lead to an infinite loop if the data structure has a circular reference. To avoid this, you should add a base case that stops the recursion when a certain condition is met.